aiida_mlip/data/model.py

```diff
@@ -231,32 +190,44 @@ def download(
         )
         arch_dir = (cache_dir / architecture) if architecture else cache_dir
 
-        # cache_path = cache_dir.resolve()
         arch_path = arch_dir.resolve()
         arch_path.mkdir(parents=True, exist_ok=True)
 
-        model_name = urlparse(url).path.split("/")[-1]
-
-        file = arch_path / filename if filename else arch_path / model_name
-
-        # If file already exists, use next indexed name
-        stem = file.stem
-        i = 1
-        while file.exists():
-            i += 1
-            file = file.with_stem(f"{stem}_{i}")
+        file = arch_path / filename
 
         # Download file
-        request.urlretrieve(url, file)
-
-        if force_download:
-            print(f"filename changed to {file}")
-            return cls.local_file(file=file, architecture=architecture)
-
-        # Check if the hash of the just downloaded file matches any other file
-        filepath = cls._check_existing_file(file)
-
-        return cls.local_file(file=filepath, architecture=architecture)
+        request.urlretrieve(uri, file)
+
+        model = cls.from_local(file=file, architecture=architecture)
+
+        if keep_file:
+            return model
+
+        file.unlink(missing_ok=True)
+
+        qb = QueryBuilder()
+        qb.append(ModelData, project=["attributes", "pk", "ctime"])
+
+        # Looking for ModelData in the whole database
+        for i in qb.iterdict():
+            # If the hash is the same as the new model, but not the creation time
+            # it means that there is already a model that is the same, use that
+            if (
+                "model_hash" in i["ModelData_1"]["attributes"]
+                and i["ModelData_1"]["attributes"]["model_hash"] == model.model_hash
+                and i["ModelData_1"]["attributes"]["architecture"] == model.architecture
+            ):
+                if i["ModelData_1"]["ctime"] != model.ctime:
```

Comment:
Is this 2 `if`s, not an extra `and` for a reason?

Reply:
not really, I can change it. And yeah should have filtering I guess